### DISCUSSION TITLE
Can't emulate basic "global hotkey to paste second-most-recent item" functionality

### DISCUSSION THREAD

**Discussion by TuggyNE** [hluk/CopyQ#1841] [Q&A] [answered]
CopyQ is a lot more sophisticated than other clipboard managers I've used, but I can't get it to do the single most basic building block of my clipboard workflow: use a global hotkey to paste what I had just before the current clipboard (usually, though not necessarily, switching it to also be the new current clipboard). (Fortunately, I _have_ been able to do the second-most important thing: ignore password managers.)

I looked at #971, but the first command there doesn't work at all, since it always focuses the CopyQ window instead of whatever window I used the hotkey from, which means it _adds_ a new line to the history (rather than simply moving an existing one around), and doesn't actually do any useful pasting. (It's also technically picking the second-to-oldest item, which is a bizarre interpretation of "second-to-last", but easy to fix by using index 1).

I'm using macOS Catalina, and CopyQ is allowed to do accessibility things with other windows: double-clicking on clipboards will correctly paste as expected. Restarting CopyQ doesn't fix anything.

**Comment by hluk** [comment, marked as answer]
> I looked at #971, but the first command there doesn't work at all, since it always focuses the CopyQ window instead of whatever window I used the hotkey from

That's strange, the command should not show the main application window. If the main window is already visible, you can add `hide()` or `focusPrevious()` before the `select()` line.

**Reply by TuggyNE** [reply]
I also made some tweaks to the code in case the JS auto-semicolon insertion wasn't quite doing what I expected. Final version, which now works:

```javascript
copyq:
    focusPrevious()
    var numberOfItems = count()
    if (numberOfItems < 2) abort()
    
    select(1)
    paste()
```
